migrations/20231210123554_init.sql

```diff
@@ -0,0 +1,314 @@
+CREATE TABLE event
+(
+    id             UUID PRIMARY KEY DEFAULT gen_random_uuid(),
+    -------------------------------------------------------
+    name           VARCHAR(255) NOT NULL,
+    description    TEXT,
+    website        VARCHAR(255),
+    accepts_staff  BOOLEAN NOT NULL,
+    start_date     DATE NOT NULL,
+    end_date       DATE NOT NULL,
+    avatar_url     VARCHAR(255),
+    -------------------------------------------------------
+    created_at     TIMESTAMP NOT NULL DEFAULT now(),
+    edited_at      TIMESTAMP NOT NULL DEFAULT now(),
+    deleted_at     TIMESTAMP,
+    -------------------------------------------------------
+    CONSTRAINT check_event_name_len
+        CHECK (char_length(name) >= 1),
+    CONSTRAINT check_event_start_date_lte_end_date
+        CHECK (start_date >= end_date),
+    CONSTRAINT check_event_created_at_lte_edited_at
+        CHECK (edited_at >= created_at)
+);
+
+
+CREATE TABLE timesheet
+(
+    id           UUID PRIMARY KEY DEFAULT gen_random_uuid(),
+    -------------------------------------------------------
+    start_date   DATE NOT NULL,
+    end_date     DATE NOT NULL,
+    total_hours  hours_per_month_float NOT NULL,
+    is_editable  BOOLEAN NOT NULL,
+    manager_note TEXT,
+    -------------------------------------------------------
+    created_at   TIMESTAMP NOT NULL DEFAULT now(),
+    edited_at    TIMESTAMP NOT NULL DEFAULT now(),
+    deleted_at   TIMESTAMP,
+    --------------------------------------------------------
+    user_id      UUID,
+    company_id   UUID,
+    event_id     UUID,
+    --------------------------------------------------------
+    FOREIGN KEY  (user_id) REFERENCES "user" (id),
+    FOREIGN KEY  (company_id) REFERENCES company (id),
+    FOREIGN KEY  (event_id) REFERENCES event (id),
+    --------------------------------------------------------
+    CONSTRAINT check_timesheet_start_date_lte_end_date
+        CHECK (start_date >= end_date),
+    CONSTRAINT check_timesheet_created_at_lte_edited_at
+        CHECK (edited_at >= created_at)
+);
+
+
+CREATE TABLE work_day
+(   
+    date         DATE NOT NULL,
+    total_hours  hours_per_day_float NOT NULL,
+    comment      TEXT,
+    is_editable  BOOLEAN NOT NULL,
+    --------------------------------------------------------
+    created_at   TIMESTAMP NOT NULL DEFAULT now(),
+    edited_at    TIMESTAMP NOT NULL DEFAULT now(),
+    deleted_at   TIMESTAMP,
+    --------------------------------------------------------
+    timesheet_id UUID,
+    --------------------------------------------------------
+    FOREIGN KEY  (timesheet_id) REFERENCES timesheet (id),
+    --------------------------------------------------------
+    CONSTRAINT check_work_day_created_at_lte_edited_at
+        CHECK (edited_at >= created_at)
+);
+
+
+CREATE TABLE event_staff
+(
+    id          UUID PRIMARY KEY DEFAULT gen_random_uuid(),
+    -------------------------------------------------------
+    level       employee_level NOT NULL,
```

Comment:
This is the incorrect type. This should not be employee_level, employee_level is used in the employment table.
This should be the event_role type, which is used for keeping track of staff privilege levels within a given event.